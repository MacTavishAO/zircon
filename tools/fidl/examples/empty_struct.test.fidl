// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

library fidl.examples.emptystruct;

struct Empty {
};

[ForDeprecatedCBindings]
protocol EmptyProtocol {
    Send(Empty e);
    -> Receive(Empty e);
    SendAndReceive(Empty e) -> (Empty e);
};
