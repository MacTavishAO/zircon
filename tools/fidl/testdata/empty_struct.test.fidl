library fidl.test.json;

struct Empty {
};

protocol EmptyProtocol {
    Send(Empty e);
    -> Receive(Empty e);
    SendAndReceive(Empty e) -> (Empty e);
};
