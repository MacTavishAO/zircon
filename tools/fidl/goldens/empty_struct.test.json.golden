{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "fidl.test.json/EmptyProtocol",
      "location": {
        "filename": "empty_struct.test.fidl",
        "line": 6,
        "column": 10,
        "length": 13
      },
      "methods": [
        {
          "ordinal": 550167292114688515,
          "name": "Send",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 7,
            "column": 5,
            "length": 4
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/Empty",
                "nullable": false
              },
              "name": "e",
              "location": {
                "filename": "empty_struct.test.fidl",
                "line": 7,
                "column": 16,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 7
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": false,
          "is_composed": false
        },
        {
          "ordinal": 7186107129703123093,
          "name": "Receive",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 8,
            "column": 8,
            "length": 7
          },
          "has_request": false,
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/Empty",
                "nullable": false
              },
              "name": "e",
              "location": {
                "filename": "empty_struct.test.fidl",
                "line": 8,
                "column": 22,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 7
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix1",
          "maybe_response_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": false
        },
        {
          "ordinal": 2187042876758414869,
          "name": "SendAndReceive",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 9,
            "column": 5,
            "length": 14
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/Empty",
                "nullable": false
              },
              "name": "e",
              "location": {
                "filename": "empty_struct.test.fidl",
                "line": 9,
                "column": 26,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 7
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix2",
          "maybe_request_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/Empty",
                "nullable": false
              },
              "name": "e",
              "location": {
                "filename": "empty_struct.test.fidl",
                "line": 9,
                "column": 39,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 7
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix3",
          "maybe_response_type_shape_v1": {
            "inline_size": 24,
            "alignment": 8,
            "depth": 0,
            "max_handles": 0,
            "max_out_of_line": 0,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": false
        }
      ]
    }
  ],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "empty_struct.test.fidl",
        "line": 7,
        "column": 9,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/Empty",
            "nullable": false
          },
          "name": "e",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 7,
            "column": 16,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 7
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix1",
      "location": {
        "filename": "empty_struct.test.fidl",
        "line": 8,
        "column": 15,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/Empty",
            "nullable": false
          },
          "name": "e",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 8,
            "column": 22,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 7
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix2",
      "location": {
        "filename": "empty_struct.test.fidl",
        "line": 9,
        "column": 19,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/Empty",
            "nullable": false
          },
          "name": "e",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 9,
            "column": 26,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 7
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix3",
      "location": {
        "filename": "empty_struct.test.fidl",
        "line": 9,
        "column": 32,
        "length": 9
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/Empty",
            "nullable": false
          },
          "name": "e",
          "location": {
            "filename": "empty_struct.test.fidl",
            "line": 9,
            "column": 39,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 7
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 8,
        "alignment": 8,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/Empty",
      "location": {
        "filename": "empty_struct.test.fidl",
        "line": 3,
        "column": 8,
        "length": 5
      },
      "anonymous": false,
      "members": [],
      "type_shape_v1": {
        "inline_size": 1,
        "alignment": 1,
        "depth": 0,
        "max_handles": 0,
        "max_out_of_line": 0,
        "has_padding": false,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/Empty",
    "fidl.test.json/EmptyProtocol"
  ],
  "declarations": {
    "fidl.test.json/EmptyProtocol": "interface",
    "fidl.test.json/SomeLongAnonymousPrefix0": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix1": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix2": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix3": "struct",
    "fidl.test.json/Empty": "struct"
  }
}
